My bad, I'll make it so that it checks `stdenv.isLinux` - what I intended to express is that nextcloud-client tries to find inotify at build time and uses it if it's available.

Happily I'm at my workplace right now where I have access to a Mac to build on, and the build fails with

```
make[2]: Entering directory '/private/var/folders/jt/qgrx8lp10ylbdlnb93t54r6c0000gn/T/nix-build-alsa-lib-1.1.4.1.drv-0/alsa-lib-1.1.4.1/src/control'
  CC       cards.lo
In file included from cards.c:35:
In file included from ./control_local.h:22:
../../include/local.h:47:2: error: Header defining endianness not defined
#error Header defining endianness not defined
 ^
1 error generated.
make[2]: *** [Makefile:334: cards.lo] Error 1
make[2]: Leaving directory '/private/var/folders/jt/qgrx8lp10ylbdlnb93t54r6c0000gn/T/nix-build-alsa-lib-1.1.4.1.drv-0/alsa-lib-1.1.4.1/src/control'
make[1]: *** [Makefile:478: all-recursive] Error 1
make[1]: Leaving directory '/private/var/folders/jt/qgrx8lp10ylbdlnb93t54r6c0000gn/T/nix-build-alsa-lib-1.1.4.1.drv-0/alsa-lib-1.1.4.1/src'
make: *** [Makefile:338: all-recursive] Error 1
builder for ‘/nix/store/9djjg7idiqp7awyx6v3568fkykacrs3l-alsa-lib-1.1.4.1.drv’ failed with exit code 2
fetching path ‘/nix/store/3ip043cbchgj6m4hjfiwn0qw87jdcaz0-qtbase-5.6.2-dev’...
cannot build derivation ‘/nix/store/2chdyipixw3ncqisy3316w98vrwaj9m5-qtmultimedia-5.6.2.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/z6yl0n1s54ny3zfqz660kfx14w8sn3l0-qtlocation-5.6.2.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/24axc64rwn91h8yk8hv8b6vlg2r7jgqz-qtwebkit-5.6.2.drv’: 1 dependencies couldn't be built
killing process 7940
cannot build derivation ‘/nix/store/hn2r0a5186ygw93bxzwh7w74y5v0ik9i-nextcloud-client-2.3.2.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/hn2r0a5186ygw93bxzwh7w74y5v0ik9i-nextcloud-client-2.3.2.drv’ failed
nix-build -A nextcloud-client -I nixpkgs=.  5.87s user 4.06s system 54% cpu 18.223 total
```

Which is a dependency of `qtmultimedia` and was not introduced by this commit.

[I'm certain this can build on Darwin](https://github.com/owncloud/client/blob/master/doc/building.rst#mac-os-x), but I think that's out of scope for this PR.

--- a/pkgs/applications/networking/nextcloud-client/default.nix
+++ b/pkgs/applications/networking/nextcloud-client/default.nix
@@ -16,6 +16,8 @@ stdenv.mkDerivation rec {
   buildInputs = [ qtbase qtwebkit qtkeychain sqlite ]
     ++ stdenv.lib.optional stdenv.isLinux inotify-tools;
 
+  enableParallelBuilding = true;
+
   dontUseCmakeBuildDir = true;
 
   cmakeDir = "client";

--- a/pkgs/applications/networking/nextcloud-client/default.nix
+++ b/pkgs/applications/networking/nextcloud-client/default.nix
@@ -1,4 +1,5 @@
-{ stdenv, fetchgit, cmake, pkgconfig, qtbase, qtwebkit, qtkeychain, sqlite }:
+{ stdenv, fetchgit, cmake, pkgconfig, qtbase, qtwebkit, qtkeychain, sqlite
+, inotify-tools }:
 
 stdenv.mkDerivation rec {
   name = "nextcloud-client-${version}";
@@ -12,7 +13,8 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [ pkgconfig cmake ];
-  buildInputs = [ qtbase qtwebkit qtkeychain sqlite ];
+  buildInputs = [ qtbase qtwebkit qtkeychain sqlite ]
+    ++ stdenv.lib.optional stdenv.isLinux inotify-tools;
 
   dontUseCmakeBuildDir = true;
 
@@ -22,6 +24,9 @@ stdenv.mkDerivation rec {
     "-UCMAKE_INSTALL_LIBDIR"
     "-DCMAKE_BUILD_TYPE=Release"
     "-DOEM_THEME_DIR=${src}/nextcloudtheme"
+  ] ++ stdenv.lib.optionals stdenv.isLinux [
+    "-DINOTIFY_LIBRARY=${inotify-tools}/lib/libinotifytools.so"
+    "-DINOTIFY_INCLUDE_DIR=${inotify-tools}/include"
   ];
 
   meta = with stdenv.lib; {

--- a/pkgs/applications/networking/nextcloud-client/default.nix
+++ b/pkgs/applications/networking/nextcloud-client/default.nix
@@ -1,5 +1,5 @@
 { stdenv, fetchgit, cmake, pkgconfig, qtbase, qtwebkit, qtkeychain, sqlite
-, inotify-tools }:
+, inotify-tools, withGnomeKeyring ? false, makeWrapper, libgnome_keyring }:
 
 stdenv.mkDerivation rec {
   name = "nextcloud-client-${version}";
@@ -13,8 +13,10 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [ pkgconfig cmake ];
+
   buildInputs = [ qtbase qtwebkit qtkeychain sqlite ]
-    ++ stdenv.lib.optional stdenv.isLinux inotify-tools;
+    ++ stdenv.lib.optional stdenv.isLinux inotify-tools
+    ++ stdenv.lib.optional withGnomeKeyring makeWrapper;
 
   enableParallelBuilding = true;
 
@@ -30,6 +32,11 @@ stdenv.mkDerivation rec {
     "-DINOTIFY_LIBRARY=${inotify-tools}/lib/libinotifytools.so"
     "-DINOTIFY_INCLUDE_DIR=${inotify-tools}/include"
   ];
+
+  postInstall = stdenv.lib.optionalString (withGnomeKeyring) ''
+    wrapProgram "$out/bin/nextcloud" \
+      --prefix LD_LIBRARY_PATH : ${stdenv.lib.makeLibraryPath [ libgnome_keyring ]}
+  '';
 
   meta = with stdenv.lib; {
     description = "Nextcloud themed desktop client";

